### ISSUE TITLE
The problem of shape mismatch in data training

### ISSUE THREAD

**Issue by 946548737619** [baaivision/UniVLA#46]

**Comment by Robertwyq** [comment]
@946548737619 
Sorry for the late reply.

Based on the first image, it seems that some sequences are of length 32 and others are 20, but according to the preset training script, it should be 2*10. The 32-length sequences seem strange. You might check if the value passed as T to the random_frames_to_tensor function is set to 20.

<img width="932" height="308" alt="Image" src="https://github.com/user-attachments/assets/06c73657-f33f-4736-9f56-66597e3aebdf" />

**Comment by 946548737619** [comment]
> [@946548737619](https://github.com/946548737619) Sorry for the late reply.
> 
> Based on the first image, it seems that some sequences are of length 32 and others are 20, but according to the preset training script, it should be 2*10. The 32-length sequences seem strange. You might check if the value passed as T to the random_frames_to_tensor function is set to 20.
> 
> <img alt="Image" width="932" height="308" src="https://private-user-images.githubusercontent.com/39329846/531817158-06c73657-f33f-4736-9f56-66597e3aebdf.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Njc5NTE2MjksIm5iZiI6MTc2Nzk1MTMyOSwicGF0aCI6Ii8zOTMyOTg0Ni81MzE4MTcxNTgtMDZjNzM2NTctZjMzZi00NzM2LTlmNTYtNjY1OTdlM2FlYmRmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjAxMDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwMTA5VDA5MzUyOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTMyMDRhZDAwZDZkMDc2N2E5NjI2ZTZjY2VjODJiZWYzMDg0MTA4N2Y1MmRiNGUyYWEzMGVhYTRjZjczMzc4ZjMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.aV1ai64AwlTkGjkec8W4PVcarP6hU8_BsxJ3q9dueI0">






Thank you for your reply. May I ask if you trained the data of the four tasks together or separately when training libero? How long did you train libero with action Fine-tuning and how many cards were used? It seems that the paper only reports the training time after the world model but does not include the time for action fine-tuning. Although I haven't successfully reproduced it yet, seeing that someone in the problem area has successfully reproduced it, I will keep trying. Thank you for your excellent work.

Currently, I have trained 8 H200s for individual libero-10/long tasks. The training time and video memory are as follows. I'm not sure if it's normal.

<img width="1696" height="666" alt="Image" src="https://github.com/user-attachments/assets/d7bb96ca-769d-4647-a2b8-8dd5be010153" />

<img width="1253" height="926" alt="Image" src="https://github.com/user-attachments/assets/df513a7e-484d-46fa-8699-81a82c073375" />

**Comment by Robertwyq** [comment]
@946548737619 
The training environment is kept consistent with EMU3 (https://github.com/baaivision/Emu3) and can directly follow its setup.

For LIBERO, the model was trained using 24 GPUs; however, the loss being zero is not normal.

**Comment by Robertwyq** [comment]
@946548737619 
Sorry, I may not check the issues very frequently. If you need more information, feel free to contact me via [email](yuqi.robert@gmail.com).
